src/logmessage.cpp

```diff
@@ -108,7 +115,19 @@ namespace g3 {
       return out;
    }
 
-
+   std::string LogMessage::message() const {
+      std::string msg;
+      if (!_wmessage.empty())
+      {
+#if !defined(__GNUC__) || (__GNUC__ >= 5)
+         std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> convert;
+         msg.append(_message.append(convert.to_bytes(_wmessage)));
```

Comment:
I think we can skip the compilation test but have a CMAKE generated `#define` (please see how the options are handled). 

1. As long as it is the sink that deals with the cpu load I think it is fine

2. Why not splitting out the functionality into a ```wmessage()``` function. It could be something like
```
bool hasWMessage(){ ... }
std::string wmessage() { .... }
```
That way a sink can easily check if the message struct contains a `wstring` and run the ```wmessage()``` function which would then contain the `convert` functionality you currently have in the ```std::string message() {...}``` function. 

3. If you make the convert object a ```std::thread_local``` then it will be thread safe and it will only be instantiated once.